src/Symfony/Component/Translation/Bridge/Loco/LocoProvider.php

```diff
@@ -93,19 +96,44 @@ public function read(array $domains, array $locales): TranslatorBag
 
         foreach ($locales as $locale) {
             foreach ($domains as $domain) {
+                $previousCatalogue = $this->translatorBag?->getCatalogue($locale);
+
                 // Loco forbids concurrent requests, so the requests must be synchronous in order to prevent "429 Too Many Requests" errors.
                 $response = $this->client->request('GET', sprintf('export/locale/%s.xlf', rawurlencode($locale)), [
                     'query' => [
                         'filter' => $domain,
                         'status' => 'translated,blank-translation',
                     ],
+                    'headers' => [
+                        'If-Modified-Since' => $previousCatalogue instanceof CatalogueMetadataAwareInterface ? $previousCatalogue->getCatalogueMetadata('last-modified', $domain) : null,
+                    ],
                 ]);
 
                 if (404 === $response->getStatusCode()) {
                     $this->logger->warning(sprintf('Locale "%s" for domain "%s" does not exist in Loco.', $locale, $domain));
                     continue;
                 }
 
+                if (304 === $response->getStatusCode()) {
+                    $this->logger->info(sprintf('No modifications found for locale "%s" and domain "%s" in Loco.', $locale, $domain));
+
+                    $catalogue = new MessageCatalogue($locale);
+
+                    foreach ($previousCatalogue->all($domain) as $key => $message) {
```

Comment:
broken if `$previousCatalogue` is null because there is no translator bag, isn't it?

Reply:
Oh yeah, that's why I've added the `null !== $this->translatorBag` in the previous `if` but I didn't remember why.

Would be more readable if I check for `$previousCatalogue` instead.

Thanks

Reply:
the API cannot return a 304 is $previousCatalogue is null

Reply:
That's true, this case should not happens since we only have a 304 if we send the last-modified date from a previous catalogue.

However, Psalm may not like this

Reply:
Psalm is not a god :godmode:

Reply:
That's true, it reminds me a conference from Fabbien IIRC :stuck_out_tongue:

Reply:
:brain: